blocks_vertical/data.js

```diff
@@ -0,0 +1,167 @@
+'use strict';
+
+goog.provide('Blockly.Blocks.data');
+
+goog.require('Blockly.Blocks');
+
+goog.require('Blockly.Colours');
+
+goog.require('Blockly.constants');
+
+Blockly.Blocks['data_variablemenu'] = {
+  /**
+   * Variable menu.
+   * @this Blockly.Block
+   */
+  init: function() {
+    this.jsonInit(
+      {
+        "message0": "%1",
+        "args0": [
+          {
+            "type": "field_variable",
+            "name": "VARIABLE",
+            "options": [
+              ['v', 'V'],
+              ['v2', 'V2'],
+              ['v3', 'V3'],
+              ['v4', 'V4'],
+              ['v5', 'V5'],
+              ['v6', 'V6'],
+              ['v7', 'V7'],			  
+            ]
+          }
+        ],
+        "inputsInline": true,
+        "output": "String",
+        "colour": Blockly.Colours.data.secondary,
+        "colourSecondary": Blockly.Colours.data.secondary,
+        "colourTertiary": Blockly.Colours.data.tertiary,
+        "outputShape": Blockly.OUTPUT_SHAPE_ROUND
+      });
+  }
+};
+
+Blockly.Blocks['data_variable'] = {
+  /**
+   * Block of Variables
+   * @this Blockly.Block
+   */
+  init: function() {
+    this.jsonInit({
+      "message0": "%1",
+        "args0": [
+          {
+            "type": "input_value",
+            "name": "VARIABLE"
+          }
+        ],  
+      "colour": Blockly.Colours.data.primary,
+      "colourSecondary": Blockly.Colours.data.secondary,
+      "colourTertiary": Blockly.Colours.data.tertiary,
+      "output": "Number",

```

Comment:
I think the intention is to keep the behavior the same as in 2.0 with this sort of thing. In fact maybe the right output type here is "Any." We'll make sure the behavior is right when we come around to implementing variables in the VM :) Thanks for pointing this out - it will definitely be helpful then.